test/automl/test_forecast.py

```diff
@@ -572,10 +576,140 @@ def smape(y_pred, y_test):
     print(automl.min_resource)
 
 
+def test_cv_step():
+    n = 300
+    time_col = "date"
+    df = pd.DataFrame(
+        {
+            time_col: pd.date_range(start="1/1/2001", periods=n, freq="D"),
+            "y": np.sin(np.linspace(start=0, stop=200, num=n)),
+        }
+    )
+
+    def split_by_date(df: pd.DataFrame, dt: datetime.date):
+        dt = datetime.datetime(dt.year, dt.month, dt.day)
+        return df[df[time_col] <= dt], df[df[time_col] > dt]
+
+    horizon = 60
+    data_end = df.date.max()
+    train_end = data_end - datetime.timedelta(days=horizon)
+
+    train_df, val_df = split_by_date(df, train_end)
+    from flaml import AutoML
+
+    tgts = ["y"]
+    # tgt = "SERIES_SANCTIONS"
+
+    preds = {}
+    for tgt in tgts:
+        features = []  # [c for c in train_df.columns if "SERIES" not in c and c != time_col]
+
+        automl = AutoML(time_budget=60, metric="mae", task="ts_forecast", eval_method="cv")
```

Comment:
Is it necessary to run the test for this long? We want to keep test time as short as having enough coverage.

Reply:
Duration reset to 5s